dashboard/scripts/archive/backfill_script_level_ids_for_teacher_feedbacks_2.rb

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env ruby
+# Backfill existing TeacherFeedbacks to set script_level_id based on user progress.
+
+require_relative '../../config/environment'
+
+def feedbacks_without_script_level_id
+  TeacherFeedback.where(script_level_id: nil)
+end
+
+def puts_count
+  puts "There are #{feedbacks_without_script_level_id.count} feedbacks without a script_level_id"
+end
+
+def update_script_level_ids
+  puts "backfilling script_level_ids..."
+  feedbacks_without_script_level_id.find_each do |feedback|
+    associated_script_levels = feedback.level.script_levels
+    if associated_script_levels.length > 1
+      associated_user_levels = UserLevel.where(
+        user_id: feedback.student_id,
+        level_id: feedback.level_id
+      )
+      if associated_user_levels.length == 1
+        script_level_id = ScriptLevel.where(
+          level_id: associated_user_levels[0].level_id,
+          script_id: associated_user_levels[0].script_id
+        ).id
+        feedback.update_attributes(script_level_id: script_level_id)
```

Comment:
`ScriptLevel.where` will return an `ActiveRecord::Relation`, not a single record.  You'll probably want to check again that there's only one, before using it to update the feedback script_level_id.

Reply:
updated